### ISSUE TITLE
Unable to build llamacpp-server

### ISSUE THREAD

**Issue by cedricclyburn** [containers/podman-desktop-extension-ai-lab#923]
Hi folks!

I'm using the Summarizer demo recipe but am unable to get things started. It looks like I have no trouble downloading the source code and model, but unfortunately, I can't get the llamacpp server started. This is on Apple Silicon, Sonoma 14.4.1, Podman Desktop v1.9.1 with Podman 5.0.2, this error also happened on my Podman 4.9.2 that I originally had before updating.

Getting this error: `no image found for llamacpp-server:latest`

<img width="1162" alt="Screenshot 2024-04-17 at 10 41 15 PM" src="https://github.com/containers/podman-desktop-extension-ai-lab/assets/54345140/7297adb9-a549-407a-986b-e041588a7843">

<img width="1162" alt="Screenshot 2024-04-17 at 10 41 46 PM" src="https://github.com/containers/podman-desktop-extension-ai-lab/assets/54345140/62ff0698-cba0-42b6-924c-7e9f612783bd">

Logs:
```
main ↪️ [ai-lab] Reading configuration from /Users/cedric/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/summarizer/summarizer-langchain/ai-lab.yaml.
main ↪️ [ai-lab] There is no workers compatible. Using default local mounting
main ↪️ [ai-lab] Application Manager using context /Users/cedric/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/summarizer/playground for container llamacpp-server
main ↪️ [ai-lab] Application Manager using context /Users/cedric/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/summarizer/summarizer-langchain for container streamlit-summary-app
```

**Comment by cedricclyburn** [comment]
Full logs:
<details>
warn : MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 extensions-started listeners added. Use emitter.setMaxListeners() to increase limit
warn : MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 extension-started listeners added. Use emitter.setMaxListeners() to increase limit
warn : MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 extension-stopped listeners added. Use emitter.setMaxListeners() to increase limit
log : main ↪️ PluginSystem: received dom-ready event from the UI
log : main ↪️ Starting http server to handle webviews on port 44000
log : main ↪️ Error while fetching API groups: Error: No active cluster!
log : main ↪️ Error while fetching API groups: Error: No active cluster!
log : main ↪️ Delayed startup, flushing 10 events
log : main ↪️ System ready. Loading extensions...
debug : main ↪️ skip watching deployments in context dev-sandbox-context-devnexus, as the context is not reachable
debug : main ↪️ skip watching services in context dev-sandbox-context-devnexus, as the context is not reachable
debug : main ↪️ skip watching ingresses in context dev-sandbox-context-devnexus, as the context is not reachable
debug : main ↪️ skip watching routes in context dev-sandbox-context-devnexus, as the context is not reachable
log : main ↪️ Activating extension (podman-desktop.compose) with max activation time of 10 seconds
log : main ↪️ Activating extension (podman-desktop.docker) with max activation time of 10 seconds
log : main ↪️ Activating extension (podman-desktop.kind) with max activation time of 10 seconds
log : main ↪️ Activating extension (podman-desktop.kube-context) with max activation time of 10 seconds
log : main ↪️ [kube-context] starting extension kube-context
log : main ↪️ Activating extension (podman-desktop.kubectl-cli) with max activation time of 10 seconds
log : main ↪️ Activating extension (podman-desktop.lima) with max activation time of 10 seconds
debug : main ↪️ [lima] Could not find socket at /Users/cedric/.lima/podman/sock/podman.sock
log : main ↪️ Activating extension (podman-desktop.podman) with max activation time of 10 seconds
log : main ↪️ Activating extension (podman-desktop.registries) with max activation time of 10 seconds
log : main ↪️ Activating extension (redhat.ai-lab) with max activation time of 10 seconds
log : main ↪️ [ai-lab] starting AI Lab extension
log : main ↪️ Activating extension (podman-desktop.minikube) with max activation time of 10 seconds
log : main ↪️ Activating extension (redhat.redhat-sandbox) with max activation time of 10 seconds
log : main ↪️ [redhat-sandbox] starting extension openshift-sandbox
log : main ↪️ [redhat-sandbox] Configfile location /Users/cedric/.kube/config
log : main ↪️ Activating extension (redhat.podman-desktop-demo) with max activation time of 10 seconds
log : main ↪️ [podman-desktop-demo] starting extension podman desktop demo
log : main ↪️ Activating extension (podman-desktop.docker) ended in 1087 milliseconds
log : main ↪️ Activating extension (podman-desktop.lima) ended in 1042 milliseconds
log : main ↪️ Activating extension (podman-desktop.registries) ended in 1029 milliseconds
log : main ↪️ Activating extension (redhat.redhat-sandbox) ended in 27 milliseconds
log : main ↪️ Activating extension (redhat.podman-desktop-demo) ended in 20 milliseconds
log : main ↪️ Activating extension (podman-desktop.compose) ended in 1116 milliseconds
error : main ↪️ Unable to fetch the available extensions: Timeout awaiting 'lookup' for 1000ms
error : main ↪️ [compose] Error getting compose version: Error: Failed to execute command: spawn /Users/cedric/.local/share/containers/podman-desktop/extensions-storage/podman-desktop.compose/bin/docker-compose ENOENT
log : main ↪️ [kind] kind extension is active
log : main ↪️ Activating extension (podman-desktop.kind) ended in 1262 milliseconds
log : main ↪️ Activating extension (podman-desktop.kubectl-cli) ended in 1246 milliseconds
log : main ↪️ [kind] kind extension is active
log : main ↪️ [kind] kind extension is active
log : main ↪️ [minikube] minikube extension is active
log : main ↪️ Activating extension (podman-desktop.minikube) ended in 561 milliseconds
log : main ↪️ Activating extension (podman-desktop.kube-context) ended in 1285 milliseconds
log : main ↪️ [ai-lab] updated indexHtml to <!DOCTYPE html>
<html class="fixed" lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" href="http://ad4d15dd-57fb-44d9-b8c1-8cc009982689.webview.localhost:44000/media/favicon.ico" />
    <meta name="viewport" content="width=device-width, height=device-height, initial-scale=1.0" />
    <title>Podman Desktop</title>
    <script type="module" crossorigin src="http://ad4d15dd-57fb-44d9-b8c1-8cc009982689.webview.localhost:44000/media/index-DNHUpfRA.js"></script>
    <link rel="stylesheet" crossorigin href="http://ad4d15dd-57fb-44d9-b8c1-8cc009982689.webview.localhost:44000/media/index-B2G80Lpv.css">
  </head>
  <body class="overflow-hidden text-white">
    <div id="app"></div>
  </body>
</html>

log : main ↪️ Activating extension (redhat.ai-lab) ended in 649 milliseconds
log : main ↪️ [kind] kind extension is active
log : main ↪️ [minikube] minikube extension is active
log : main ↪️ [kind] kind extension is active
log : main ↪️ [minikube] minikube extension is active
log : main ↪️ [kind] kind extension is active
log : main ↪️ [minikube] minikube extension is active
log : main ↪️ [kind] kind extension is active
log : main ↪️ [minikube] minikube extension is active
log : main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 288.5595420598984ms
warn : main ↪️ [podman]  Can't get machine podman-machine-default resource usage error Error: no engine matching this container
log : main ↪️ Activating extension (podman-desktop.podman) ended in 1431 milliseconds
log : main ↪️ PluginSystem: initialization done.
log : main ↪️ Autostarting podman-desktop.podman container engine
log : main ↪️ [podman] Podman extension: Do not start a machine as there is already one starting or started
log : main ↪️ [kind] kind extension is active
log : main ↪️ [minikube] minikube extension is active
log : main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 43.16100001335144ms
log : main ↪️ Skipping update for extension redhat.ai-lab because installed version 0.4.0 is greater or equal to latest available version 0.4.0
log : main ↪️ Skipping update for extension podman-desktop.minikube because installed version 0.3.0-next is greater or equal to latest available version 0.2.0
log : main ↪️ Skipping update for extension redhat.redhat-sandbox because installed version 0.0.2 is greater or equal to latest available version 0.0.2
log : main ↪️ Skipping update for extension redhat.podman-desktop-demo because it is not available in the registry
log : main ↪️ [ai-lab] Reading configuration from /Users/cedric/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/summarizer/summarizer-langchain/ai-lab.yaml.
warn : main ↪️ [ai-lab] There is no workers compatible. Using default local mounting
log : main ↪️ [ai-lab] Application Manager using context /Users/cedric/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/summarizer/playground for container llamacpp-server
log : main ↪️ [ai-lab] Application Manager using context /Users/cedric/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/summarizer/summarizer-langchain for container streamlit-summary-app
error : main ↪️ [ai-lab] Something went wrong while building the image:  building at STEP "RUN pip install --no-cache-dir --upgrade -r /locallm/requirements.txt": while running runtime: exit status 1

log : main ↪️ event is [object Object]
error : main ↪️ [ai-lab] Something went wrong while trying to start application Error: no image found for llamacpp-server:latest
log : main ↪️ event is [object Object]
</details>

**Comment by vrothberg** [comment]
Thanks for reaching out!

I cannot reproduce on my end. @benoitf @jeffmaury can you?

**Comment by jeffmaury** [comment]
> Thanks for reaching out!
> 
> I cannot reproduce on my end. @benoitf @jeffmaury can you?

May check but busy on 1.0.0 and this will be fixed by 1.0.0

**Comment by axel7083** [comment]
The interesting line in the logs is the following

> Something went wrong while building the image: building at STEP "RUN pip install --no-cache-dir --upgrade -r /locallm/requirements.txt": while running runtime: exit status 1

We seems not to be catching the right error, linked issue https://github.com/containers/podman-desktop-extension-ai-lab/issues/741
